src/pykiso/lib/connectors/cc_process.py

```diff
@@ -0,0 +1,265 @@
+                if self.text:
+                    data = stream.readline()
+                else:
+                    data = stream.read(1)
+                if len(data) == 0:
+                    break
+                # print(data)
+                self.queue_in.put(ProcessMessage(name, data))
+        finally:
+            with self.lock:
+                self.ready += 1
+                if self.ready == int(self.pipe_stdout) + int(self.pipe_stderr):
+                    # None marks the termination of all read threads
+                    self.queue_in.put(None)
+
+    def _cc_close(self) -> None:
+        self._cleanup()
+
+    def _cc_send(self, msg: MessageType, raw: bool = False, **kwargs) -> None:
+        """Execute process commands or write data to stdin"""
+        if isinstance(msg, dict) and "command" in msg:
+            if msg["command"] == "start":
+                self.start(msg["executable"], msg["args"])
+        elif self.pipe_stdin:
+            if self.process is None:
+                raise CCProcessError("Process is not running.")
+            log.debug(f"write stdin: {msg}")
+            self.process.stdin.write(msg)
+            self.process.stdin.flush()
+        else:
+            raise CCProcessError("Can not send to stdin because pipe is not enabled.")
+
+    def _cleanup(self):
+        """Cleanup threads and process objects"""
+        if self.process is not None:
+            self.process.terminate()
+            self.process.wait()
+        if self.stdout_thread is not None:
+            self.stdout_thread.join()
+            self.stdout_thread = None
+        if self.stderr_thread is not None:
+            self.stderr_thread.join()
+            self.stderr_thread = None
+        if self.queue_in is not None:
+            self.queue_in = None
+
+    def _cc_open(self) -> None:
+        """Implement abstract method"""
+        pass
+
+    def _read_existing(self) -> List[ProcessMessage]:
+        """Read buffered messages that where already received from the process
+
+        :return: Existing messages as tuple: (streamname, data)
+        """
+        messages = self.buffer
+
+        # Get all messages from the process that are available
+        while not self.queue_in.empty():
+            messages.append(self.queue_in.get_nowait())
+        i = 1
+        # Find messages from the same stream(first entry in the tuple) as the first message
+        while (
+            i < len(messages)
+            and messages[0] is not None
+            and messages[i] is not None
+            and messages[0].stream == messages[i].stream
+        ):
+            i += 1
+
+        # Save the remaining messages for next time
+        self.buffer = messages[i:]
+
+        # Process only messages from the same stream
+        messages = messages[:i]
+        if len(messages) == 0:
+            return []
+
+        # Join messages
+        r = [ProcessMessage(messages[0].stream, b"".join([x.data for x in messages]))]
```

Comment:
Why do you need to return a list if it has at most 1 item?

Reply:
Right, previously it returned more than one, but now thats not possible anymore. Will change that.

Reply:
Correction: It can return empty list, list with one message, list with None. When removing the list I could not return nothing.